pkg/manager/impl/execution_manager.go

```diff
@@ -490,15 +490,19 @@ func (m *ExecutionManager) getExecutionConfig(ctx context.Context, request *admi
 		return workflowExecConfig, nil
 	}
 
+	var workflowName string
 	if launchPlan != nil && launchPlan.Spec != nil {
 		// merge the launch plan spec into workflowExecConfig
 		if isChanged := mergeIntoExecConfig(workflowExecConfig, launchPlan.Spec); isChanged {
 			return workflowExecConfig, nil
 		}
+		if launchPlan.Spec.WorkflowId != nil {
+			workflowName = launchPlan.Spec.WorkflowId.Name
+		}
 	}
 
 	matchableResource, err := util.GetMatchableResource(ctx, m.resourceManager,
-		admin.MatchableResource_WORKFLOW_EXECUTION_CONFIG, request.Project, request.Domain)
+		admin.MatchableResource_WORKFLOW_EXECUTION_CONFIG, request.Project, request.Domain, workflowName)
```

Comment:
should we also pass the launch plan name here? https://github.com/flyteorg/flyteadmin/blob/master/pkg/manager/interfaces/resource.go#L35

Reply:
I see it's not used, feel free to ignore this comment as long as we capture this in an issue!